### PR TITLE
updating ssm functions to read from latest deblacklsited maf

### PR DESCRIPTION
**Pull request by mattssca**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [x] I tested the new code for my use case (please provide a reproducible example of how you tested the new functionality)

- [x] I ensured all dplyr functions that commonly conflict with other packages are fully qualified. 

This can be checked and addressed by running `check_functions.pl` and responding to the prompts. Test your code _after_ you do this.

- [x] I generated the documentation and checked for errors relating to the new function (e.g. `devtools::document()`) and added `NAMESPACE` and all other modified files in the root directory and under `man`. 

### Optional but preferred with PRs

- [ ] I updated and/or successfully knitted a vignette that relies on the modified code (which ones?)

### Required

- [x] I documented my function using [ROxygen style](https://jozef.io/r102-addin-roxytags/#:~:text=Inserting%20a%20skeleton%20%2D%20Do%20this,Shift%2BAlt%2BR%20).)

- [x] All parameters for the function are described in the documentation and the function has a decriptive title. 

Example:
```
#' Use GISTIC2.0 scores output to reproduce maftools::chromoplot with more flexibility
#'
#' @param scores output file scores.gistic from the run of GISTIC2.0
#' @param genes_to_label optional. Provide a data frame of genes to label (if mutated). The first 3 columns must contain chromosome, start, and end coordinates. Another required column must contain gene names and be named `gene`. (truncated for example)
#' @param cutoff optional. Used to determine which regions to color as aberrant. Must be float in the range [0-1]. (truncated for example)
```

- [ ] My function uses a library that isn't already a dependency of GAMBLR and I made the package aware of this dependency using the function documentation `import` statment. 

Example:
```
#' @return nothing
#' @export
#' @import tidyverse ggrepel
```

## Checklist for changes to existing code

- [x] I added/removed arguments to a function and updated documentation for all changed/new arguments

- [ ] I tested the new code for compatability with existing functionality in the Master branch (please provide a reprex of how you tested the original functionality)

### CODE REVIEW

**Comment by mattssca** [comment]
The follwoing functions have been updated to load data from the latest (and greatest) deblacklsited mafs and to filter out varaints with low read support:

get_ssm_by_sample
get_coding_ssm
get_ssm_by_region
get_ssm_by_gene (function removed and added "these_genes" parameter to get_ssm_by_sample and get_ssm_by_samples to allow for gene-subsetting).
get_coding_ssm_status
assign_cn_to_ssm (using updated get_ssm_by_samples)

The updated functions have been tested and also testing other functions that call these functions. See lsited functions bellow for such cases.

get_ssm_by_regions
calc_mutation_frequency_sliding_windows
get_ashm_count_matrix
ashm_multi_rainbow_plot - broken due to get gamblr meta
ashm_rainbow_plot
annotate_driver_ssm
estimate_purity
fancy_sv_chrdistplot
fancy_snp_chrdistplot
fancy_svbar
fancy_cnlohbar
fancy_vplot

Package documentation has also been regenerated

File: R/database.R [hunk 1]
```diff
@@ -109,32 +110,37 @@ get_ssm_by_samples = function(these_sample_ids,
...
-  if(flavour == "legacy"){
+  if(flavour=="legacy"){
     warning("I lied. Access to the old variant calls is not currently supported in this function")
     # TODO: implement loading of the old merged MAF under icgc_dart... vcf2maf-1.2 ..level_3 as per the other from_flatfile functions
     return()
-  }else if(flavour == "clustered"){
+  }else if(flavour=="clustered"){
     if(subset_from_merge){
-      maf_template = "all_the_things/slms_3-1.0_vcf2maf-1.3/{seq_type}--projection/deblacklisted/augmented_maf/all_slms-3--{projection}.maf"
+      maf_template = "all_the_things/slms_3-1.0_vcf2maf-1.3/{seq_type}--projection/deblacklisted/augmented_maf/all_slms-3--{projection}.CDS.maf"
```
Comment (rdmorin) [inline R/database.R:119]: I know I didn't do this in my version of the code but really this should be stored in the config.yml and retrieved in this code instead of using a hard-coded path

File: R/database.R [hunk 2]
```diff
@@ -109,32 +110,37 @@ get_ssm_by_samples = function(these_sample_ids,
...
   these_sample_ids = these_sample_ids[which(these_sample_ids %in% these_samples_metadata$sample_id)]
-  if(flavour == "legacy"){
+  if(flavour=="legacy"){
     warning("I lied. Access to the old variant calls is not currently supported in this function")
     # TODO: implement loading of the old merged MAF under icgc_dart... vcf2maf-1.2 ..level_3 as per the other from_flatfile functions
     return()
-  }else if(flavour == "clustered"){
+  }else if(flavour=="clustered"){
     if(subset_from_merge){
-      maf_template = "all_the_things/slms_3-1.0_vcf2maf-1.3/{seq_type}--projection/deblacklisted/augmented_maf/all_slms-3--{projection}.maf"
```
Comment (rdmorin) [inline R/database.R:118]: Why was this changed to point to the CDS file instead of the original?

**Review by rdmorin — CHANGES_REQUESTED** [review]
I'm confused by the first change so maybe we need to go over the requirements a bit more. Let's touch base on Slack.

File: R/utilities.R [hunk 3]
```diff
@@ -936,12 +961,12 @@ collate_ssm_results = function(sample_table,
   if(from_flatfile){
     base_path = config::get("project_base")
     #test if we have permissions for the full gambl + icgc merge
```
Comment (rdmorin) [inline R/utilities.R:963]: This needs to be updated. These are referring to the old outputs. NEeds to be switched to the new ones, e.g.

maf_template = config::get("results_flatfiles")$ssm$template$cds$deblacklisted

**Review by rdmorin — CHANGES_REQUESTED** [review]
I noted one function that needs updating still and one that I can't tell exactly what's changed. The rest looks great

File: R/utilities.R [hunk 4]
```diff
@@ -132,28 +138,47 @@ get_coding_ssm_status = function(gene_symbols,
...
       dplyr::filter(Variant_Classification %in% coding_class)
-
+    
   }else if (!is.null(maf_path) ){
     coding_ssm = fread_maf(maf_path)
     coding_ssm = coding_ssm %>%
       dplyr::filter(Variant_Classification %in% coding_class)
-
-  }else{
-    coding_ssm = maf_data
```
[the remark this answers is not in the document]
Comment (mattssca) [reply R/utilities.R:142]: Indeed, that could work just fine. I can add this to my list of things to do and address this in my next PR, if that's ok?

File: R/utilities.R (hunk 3 above)
Comment (mattssca) [reply R/utilities.R:963]: Indeed, good catch!

File: R/utilities.R [hunk 5]
```diff
@@ -1092,87 +1107,92 @@ assign_cn_to_ssm = function(this_sample,
...
-      seg_sample = read_tsv(battenberg_file) %>%
-        as.data.table() %>% 
-        dplyr::mutate(size = end - start) %>%
-        dplyr::filter(size > 100) %>%
-        dplyr::mutate(chrom = gsub("chr", "", chrom)) %>%
-        dplyr::rename(Chromosome = chrom, Start_Position = start, End_Position = end)
-
-      data.table::setkey(seg_sample, Chromosome, Start_Position, End_Position)
-      a = data.table::as.data.table(maf_sample)
+    message(paste("fetching:", tool_name))
```
[the remark this answers is not in the document]
Comment (mattssca) [reply R/utilities.R:1138]: So in short, the only update to this function is that it now calls the get_ssm_by_sample function. I just made another update to this function that removes the if(use_augmented_maf) statement and instead adds augmented = use_augmented_maf to the get_ssm_by_sample statement (line 1093). Previously, the function didn't handle non-augmented mafs but it does now. For all the other code that looks changed in this function, I think it's just a matter of lines being shifted, adding spaces, new lines etc. Would that make sense?

**Comment by mattssca** [comment]
Thanks for your comments Ryan, here are the two items that needs to be addressed, potentially in another PR?

1. Have get_coding_ssm_status call get_coding_ssm directly to reduce duplicated code.
2. annotate_driver_ssm still calls get_ssm_by_gene. This needs to be updated.

File: R/database.R [hunk 6]
```diff
@@ -1300,48 +1315,6 @@ get_ashm_count_matrix = function(regions_bed,
...
-#'
-#' @return MAF-format data frame of mutations in query gene.
-#' @export
-#' @import tidyverse DBI RMariaDB dbplyr
-#'
-#' @examples
-#' #basic usage
-#' ssm_ezh2 = get_ssm_by_gene(gene_symbol=c("EZH2"),coding_only=TRUE)
-#'
-get_ssm_by_gene = function(gene_symbol,
```
[the remark this answers is not in the document]
Comment (rdmorin) [reply R/database.R:1318]: Oh right. I remember why that is now. The issue is that some genes have non-coding drivers (e.g. NFKBIZ). If we insist that they supply maf_df then we can probably replace that code with something more efficient that works from the data provided by the user. Might need a quick conversation. For now can you just remove the "include_noncoding" option from that function and change maf_df to a required argument? That should prevent anyone accidentally getting the wrong results at least.

[the remark this answers is not in the document]
Comment (mattssca) [reply R/database.R:1318]: Great, I have now updated `annotate_driver_ssm` accordingly. Parameter `include_noncoding` has been removed (parameter description as well), if maf_df is not supplied, the user will receive a warning saying maf_df is required to run the function followed by return. The maf_df parameter description is also updated (removed "optional").

File: R/utilities.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (rdmorin) [reply R/utilities.R:142]: Sure

File: R/utilities.R (hunk 5 above)
[the remark this answers is not in the document]
Comment (rdmorin) [reply R/utilities.R:1131]: Yes

**Review by rdmorin — APPROVED** [review]
